bigquery_etl/cli/stage.py

```diff
@@ -260,13 +261,35 @@ def _udf_dependencies(artifact_files):
 def _view_dependencies(artifact_files, sql_dir):
     """Determine view dependencies."""
     view_dependencies = set()
-    view_dependency_files = [file for file in artifact_files if file.name == VIEW_FILE]
+    view_dependency_files = [
+        file for file in artifact_files if file.name in {VIEW_FILE, QUERY_FILE}
+    ]
     for dep_file in view_dependency_files:
         # all references views and tables need to be deployed in the same stage project
         if dep_file not in artifact_files:
             view_dependencies.add(dep_file)
 
-        if dep_file.name == VIEW_FILE:
+        if dep_file.name == QUERY_FILE:
+            # check query files for view dependencies
+            # views need to be deployed to stage as they might reference other changed artifacts upstream
+            # without getting changed themselves
+            dependencies = extract_table_references(
+                render(dep_file.name, dep_file.parent)
+            )
+            for dependency in dependencies:
+                dependency_components = dependency.split(".")
+                if len(dependency_components) == 3:
+                    # views are always fully qualified, so they'll consist of 3 parts
```

Comment:
I don't think view references are always fully qualified ([example](https://github.com/mozilla/bigquery-etl/blob/dd824d66ca2f8a1ca3f546cab758f041d7794155/sql/moz-fx-data-shared-prod/fenix_derived/events_daily_v1/query.sql#L32)).

Reply:
That is correct, I believe only inside `view.sql` we are required to use fully qualified.